cpp/src/sampling/neighborhood.cu

```diff
@@ -20,21 +20,23 @@
 
 #include <cugraph-ops/graph/sampling.hpp>
 
+#include <raft/random/rng_state.hpp>
+
 namespace cugraph {
 
 template <typename graph_t>
 std::tuple<rmm::device_uvector<typename graph_t::edge_type>,
            rmm::device_uvector<typename graph_t::vertex_type>>
 sample_neighbors_adjacency_list(raft::handle_t const& handle,
-                                ops::gnn::graph::Rng& rng,
+                                raft::random::RngState& rng_state,
```

Comment:
You will have to make the corresponding changes to cpp/include/cugraph/algorithms.hpp file. The functions sample_neighbors* do not have a python binding yet so CI should not break because of this.
@rlratzel thoghts?

Reply:
Thanks for catching this. I've updated the public API in algorithms.
I don't see any uses in python or binding code, so I think this should be good enough for now.

Reply:
> You will have to make the corresponding changes to cpp/include/cugraph/algorithms.hpp file. The functions sample_neighbors* do not have a python binding yet so CI should not break because of this. @rlratzel thoghts?

This should be okay since the python binding for neighborhood sampling will use the C API, which does not expose this function signature to the binding code.

Reply:
I will merge this change (once it is merged into the code base) into my refactor branch for uniform neighborhood sampling.  I've been talking with Corey about wanting to make this change, so I'm ready.